app/waf/src/controllers/application.controller.ts

```diff
@@ -155,4 +173,79 @@ export class ApplicationController {
   async deleteById(@param.path.string('id') id: string): Promise<void> {
     await this.applicationRepository.deleteById(id);
   }
+
+  @post(prefix + '/applications/{id}/deploy', {
+    responses: {
+      '204': {
+        description: 'Application deploy success',
+      },
+    },
+  })
+  async deployById(@param.path.string('id') id: string): Promise<Object> {
+    let application = await this.applicationRepository.findById(id);
+
+    let params: {[key: string]: Object} = {
+      application: application,
+    };
+
+    let tenantAssoc = null;
+    try {
+      tenantAssoc = await this.tenantAssociationRepository.findById(
+        application.tenantId,
+      );
+    } catch (e) {
+      if (e.code !== 'ENTITY_NOT_FOUND') {
+        throw e;
+      }
+    }
+
+    if (!tenantAssoc) {
+      try {
+        tenantAssoc = await this.tenantAssociationRepository.findById(
+          'default',
+        );
+      } catch (e) {
+        if (e.code === 'ENTITY_NOT_FOUND') {
+          throw new HttpErrors.UnprocessableEntity(
+            'No ADC associated with Application ' + application.id,
+          );
+        } else {
+          throw e;
+        }
+      }
+    }
+
+    let adc = await this.adcRepository.findById(tenantAssoc.adcId);
+    params.adc = adc;
+
+    if (application.services.length === 0) {
+      throw new HttpErrors.UnprocessableEntity(
+        'No service in Application ' + application.id,
+      );
+    } else {
+      let serviceId = application.services[0];
+      let service = await this.serviceRepository.findById(serviceId);
+
+      params.service = service;
+
+      if (service.pool) {
+        let pool = await this.poolRepository.findById(service.pool);
+        params.pool = pool;
+
+        if (pool.members.length > 0) {
+          let members = [];
+          for (let member_id of pool.members) {
+            let m = await this.memberRepository.findById(member_id);
```

Comment:
`find` can return the member list directly, thus, no `for of` is needed.

Reply:
Let's make that model to 'hasMany' memebers in near future.

Reply:
`find` style of this can be: 
```
          let members = [];
          members = await this.memberRepository.find({
            where: {
              id: {
                inq: pool.members
              }
            }
          })```